doc/userguide/rules/app-layer.rst

```diff
@@ -8,18 +8,31 @@ Match on the detected app-layer protocol.
 
 Syntax::
 
-    app-layer-protocol:[!]<protocol>;
+    app-layer-protocol:[!]<protocol>(,<mode>);
 
 Examples::
 
     app-layer-protocol:ssh;
     app-layer-protocol:!tls;
     app-layer-protocol:failed;
+    app-layer-protocol:!http,final;
+    app-layer-protocol:http,toserver; app-layer-protocol:tls,toclient;
+    app-layer-protocol:http2,final; app-layer-protocol:http1,original;
 
 A special value 'failed' can be used for matching on flows in which
 protocol detection failed. This can happen if Suricata doesn't know
 the protocol or when certain 'bail out' conditions happen.
 
+The different modes are
+* toserver : protocol recognized in the direction to server
+* toclient : protocol recognized in the direction to client
+* either : tries to match protocols found on both sides
```

Comment:
why not use "direction" instead of side, here?

Reply:
Right